Gordon360/Services/RecIM/ParticipantService.cs

```diff
@@ -0,0 +1,242 @@
+                                        EndDate = psh.EndDate
+                                    }).AsEnumerable();
+            return status;
+        }
+        public IEnumerable<TeamExtendedViewModel> GetParticipantTeams(string username)
+        {
+       
+            //to be handled by teamservice
+            var teams = _context.ParticipantTeam
+                            .Where(pt => pt.ParticipantUsername == username)
+                                .Join(_context.Team,
+                                    pt => pt.TeamID,
+                                    t => t.ID,
+                                    (pt, t) => new TeamExtendedViewModel
+                                    {
+                                        ID = t.ID,
+                                        Activity = _context.Activity.FirstOrDefault(a => a.ID == t.ActivityID),
+                                        Name = t.Name,
+                                        Status = _context.TeamStatus
+                                                    .FirstOrDefault(ts => ts.ID == t.StatusID)
+                                                    .Description,
+                                        Logo = t.Logo
+                                    });
+            return teams;
+        }
+
+        public IEnumerable<ParticipantExtendedViewModel> GetParticipants()
+        {
+            //temporary, slow and will be adjusted after we implement views in the DB
+            var participants = _context.Participant.Select(p => new ParticipantExtendedViewModel
+            {
+                Username = p.Username,
+                Email = _accountService.GetAccountByUsername(p.Username).Email,
+                Status = _context.ParticipantStatusHistory
+                                                .Where(psh => psh.ParticipantUsername == p.Username)
+                                                .OrderByDescending(psh => psh.ID)
+                                                .Take(1)
+                                                    .Join(_context.ParticipantStatus,
+                                                        psh => psh.StatusID,
+                                                        ps => ps.ID,
+                                                        (psh, ps) => ps.Description)
+                                                .FirstOrDefault(),
+                IsAdmin = p.IsAdmin
+            });
+            return participants;
+        }
+
+        public async Task<ParticipantExtendedViewModel> PostParticipantAsync(string username, int? statusID = 4)
+        {
+            await _context.Participant.AddAsync(new Participant
+            {
+                Username = username
+            });
+            await _context.ParticipantStatusHistory.AddAsync(new ParticipantStatusHistory
+            {
+                ParticipantUsername = username,
+                StatusID = statusID ?? 4, //default to cleared
```

Comment:
You should either have the default param value, or leave the param nullable and have a default statusID in the `new` expression, but not both.

Reply:
got it